engine/framework/internal/eventloop/runner.go

```diff
@@ -62,6 +62,14 @@ func (r *Runner[R]) Run(ctx context.Context) error {
 		panic(fmt.Sprintf("unexpected exiting with nil error: %s", r.task.ID()))
 	}
 
+	if isCanceledError(err) {
+		if stopErr := r.task.Stop(ctx); stopErr != nil {
+			log.Warn("Cancel task returned error",
+				zap.String("label", r.task.ID()), zap.Error(stopErr))
+		}
+		return err
```

Comment:
Why doesn't `CanceledError`  trigger `GracefulExit`?

Reply:
Because cancel is a force stop scenario, in which job master doesn't need to wait workers exit.